stun/client.go

```diff
@@ -98,12 +104,22 @@ func (c *Client) Discover() (NATType, *Host, error) {
 	conn := c.conn
 	if conn == nil {
 		var laddr *net.UDPAddr
+
+		address := ""
+		if c.localIP != "" {
+			address += c.localIP
+		}
 		if c.localPort != 0 {
-			laddr, err = net.ResolveUDPAddr("udp", fmt.Sprintf(":%d", c.localPort))
+			address += strconv.Itoa(c.localPort)
+		}
+
+		if c.localPort != 0 {
```

Comment:
why do we not put the two `if c.localPort != 0 {` together?

Reply:
I've taken care of it and look better now